docs/user-guides/plugin/development.md

```diff
@@ -0,0 +1,133 @@
+
+At the end of the guide you will have spun up the local development environment, onboarded a Cluster, created a _PluginDefinition_ and deployed it as a _Plugin_ to the onboarded Cluster.
+
+[!NOTE]
+This guide assumes you already have a working Helm chart and will not cover how to create a Helm Chart from scratch. For more information on how to create a Helm Chart, please refer to the [Helm documentation](https://helm.sh/docs/topics/charts/).
+
+## Requirements
+
+- [git](https://git-scm.com/downloads)
+- [Docker](https://docs.docker.com/engine/install/)
+- [kind](https://kind.sigs.k8s.io/docs/user/quick-start/#installation)
+- [kubectl](https://kubernetes.io/docs/reference/kubectl/)
+- [Greenhousectl](https://github.com/cloudoperators/greenhouse/releases)
+
+## Initializing the local development environment
+
+First, let's initialize the local development environment by cloning the [Greenhouse Extensions repository](https://github.com/cloudoperators/greenhouse-extensions/). This repository contains already existing Greenhouse _PluginDefinitions_ as well as the local development environment setup.
+
+### Starting the local develoment environment
+
+Follow the [Get started](https://github.com/cloudoperators/greenhouse-extensions/tree/main/dev-env#get-started) section of the dev-env README to spin up the local Greenhouse development environment.
+
+This will provide you with a local Greenhouse instance running, filled with some example Greenhouse resources and the Greenhouse UI running on `http://localhost:3000`.
+
+### Onboarding a Cluster
+
+In this step we will create and onboard a new Cluster to the local Greenhouse instance. The local cluster will be created utilizing kind.
+
+In order to onboard a kind cluster follow the [onboarding a cluster](https://github.com/cloudoperators/greenhouse-extensions/tree/main/dev-env#onboard-kind-cluster) secton of the dev-env README.
+
+After onboarding the cluster you should see the new Cluster in the Greenhouse UI.
+
+## Prepare Helm Chart
+
+For this example we will use the [bitnami nginx](https://artifacthub.io/packages/helm/bitnami/nginx) Helm Chart.
+The packaged chart can be downloaded with:
+
+```bash
+helm pull oci://registry-1.docker.io/bitnamicharts/nginx --destination ./
+```
+
+After unpacking the `*.tgz` file there is a folder named `nginx` containing the Helm Chart.
+
+## Generating a PluginDefinition from a Helm Chart
+
+Using the files of the Helm Chart we will create a new Greenhouse _PluginDefinition_ using the `greenhousectl` CLI.
+
+```bash
+greenhousectl plugin generate ./nginx ./nginx-plugin
+```
+
+This will create a new folder `nginx-plugin` containing the _PluginDefinition_ in a nested structure.
+
+## Modifying the PluginDefinition
+
+The generated _PluginDefinition_ contains a `plugin.yaml` file which defines the _PluginDefinition_. But there are still a few steps required to make it work.
```

Comment:
Should we mention the other relevant `spec` fields?
Namely:
- `icon`
- `docMarkDownUrl`
- `displayName`

Not sure how/if they are filled by `greenhousectl`?